### PR TITLE
Fix - use own ICToken instead of external ICToken

### PR DESCRIPTION
**Pull request by vminkov**
## Description

Fix - use own ICToken instead of external ICToken

### CODE REVIEW

File: contracts/compound/CToken.sol [hunk 1]
```diff
@@ -847,6 +844,24 @@ abstract contract CToken is CTokenInterface, TokenErrorReporter, Exponential, Di
...
+   * @notice Indicator that this is a CToken contract (for inspection)
+   */
+  function isCToken() external view returns (bool) {
+    return true;
+  }
+
+  /**
+   * @notice Indicator that this is or is not a CEther contract (for inspection)
+   */
+  function isCEther() external view returns (bool) {
```
Comment (carlomazzaferro) [inline contracts/compound/CToken.sol]: Let's just remove and CEther reference?

File: contracts/FuseSafeLiquidator.sol [hunk 2]
```diff
@@ -304,7 +301,7 @@ contract FuseSafeLiquidator is OwnableUpgradeable, IUniswapV2Callee {
 
   function safeLiquidate(
     address,
-    ICEther,
```
Comment (carlomazzaferro) [inline contracts/FuseSafeLiquidator.sol:307]: Remove CEther reference?

File: contracts/compound/CTokenInterfaces.sol [hunk 3]
```diff
@@ -211,176 +211,231 @@ abstract contract CTokenBaseInterface is CErc20Storage {
...
-  }
+  function isCToken() external view returns (bool);
 
   /**
    * @notice Indicator that this is or is not a CEther contract (for inspection)
    */
-  function isCEther() external virtual returns (bool) {
-    return false;
-  }
+  function isCEther() external view returns (bool);
```
Comment (carlomazzaferro) [inline contracts/compound/CTokenInterfaces.sol]: here too

File: contracts/compound/Comptroller.sol [hunk 4]
```diff
@@ -1338,14 +1330,14 @@ contract Comptroller is ComptrollerV3Storage, ComptrollerInterface, ComptrollerE
...
-  function isDeprecated(CTokenInterface cToken) public view returns (bool) {
+  function isDeprecated(ICToken cToken) public view returns (bool) {
     return
       markets[address(cToken)].collateralFactorMantissa == 0 &&
       borrowGuardianPaused[address(cToken)] == true &&
       add_(add_(cToken.reserveFactorMantissa(), cToken.adminFeeMantissa()), cToken.fuseFeeMantissa()) == 1e18;
   }
 
-  function asComptrollerFirstExtension() public view returns (ComptrollerFirstExtension) {
+  function asComptrollerExtension() internal view returns (ComptrollerFirstExtension) {
```
Comment (carlomazzaferro) [inline contracts/compound/Comptroller.sol:1340]: why internal?

Comment (vminkov) [reply contracts/compound/Comptroller.sol:1340]: the ICToken interface makes the external use of this obsolete

**Pull request merged**